hasura/graphql-engine
# persist queries in GraphiQL (closes #1894)

Fix #!894

### Description


Make GraphiQL support persisting queries if there is no `query_file` request param and there are enough tables.

### Affected components 


- Console

### Related Issues



#1894 

### Solution and Design



On GraphiQL load,
1. Check if there is `query_file` in URL parameters. If there is, populate GraphiQL with those queries.
2. If there is no `query_file` parameter, check if there are tracked tables, if there aren't show a message saying "You don't have any tables, go create some from the data section"
3. If there are tables, just use the queries from local storage.

### Steps to test and verify


1. Add a URL parameter like `?query_file=https://raw.githubusercontent.com/wawhal/test-repo/master/queries.graphql`, GraphiQL should template queries from the query file.
2. When there are no tables, GraphiQL should have a comment telling you to go create tables.
3. When there are tables, GraphiQL should load queries from local storage if there are any.

### Limitations, known bugs & workarounds
